RichText enhancements

rh101 opened the pull request:
## Describe your changes

Add paragraph tag support.

Currently, to get this output:
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Feugiat scelerisque varius morbi enim nunc. Dis parturient montes nascetur ridiculus mus mauris vitae ultricies.

Egestas integer eget aliquet nibh. Diam in arcu cursus euismod quis viverra nibh. Dictum sit amet justo donec enim diam. At urna condimentum mattis pellentesque id nibh tortor id aliquet.

We would require 4 `</br>` tags (2 at the end of each line):
```
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Feugiat scelerisque varius morbi enim nunc. Dis parturient montes nascetur ridiculus mus mauris vitae ultricies.</br>
</br>
Egestas integer eget aliquet nibh. Diam in arcu cursus euismod quis viverra nibh. Dictum sit amet justo donec enim diam. At urna condimentum mattis pellentesque id nibh tortor id aliquet.</br>
</br>
```

With paragraph tag support, the following produces the correct output:
```
<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Feugiat scelerisque varius morbi enim nunc. Dis parturient montes nascetur ridiculus mus mauris vitae ultricies.</p>
<p>Egestas integer eget aliquet nibh. Diam in arcu cursus euismod quis viverra nibh. Dictum sit amet justo donec enim diam. At urna condimentum mattis pellentesque id nibh tortor id aliquet.</p>
```

This will add 2 new line breaks at the end of the paragraph (similar to HTML renderer output).

Add HTML header tags support with default sizes (34, 30, 24, 20, 18, 16 from h1 to h6 respectively).  Header font face, size and color can be overriden via attributes:
```
<h1 face="fonts/Marker Felt.ttf" size="50">h1. HEADING</h1>
<h2>h2. HEADING</h2>
<h3>h3. HEADING</h3>
<h4>h4. HEADING</h4>
<h5>h5. HEADING</h5>
<h6>h6. HEADING</h6>
```
This will add 2 new line breaks at the end (similar to HTML renderer output).

Enhance `RichElementNewLine` to allow for more than 1 new line entries to be created, rather than having to create multiple individual `RichElementNewLine` elements.  

Add support for end of tag action.  Currently, on parsing a new tag, the parser will begin on the starting tag, and any RichText elements that need to be created are handled in this phase.  This is not ideal for all types of tags, such as the new paragraph and header tags, since they require the new line elements to be added after the text within the tags.  With the changes in this PR, this is now possible.

General code clean up.

Note that these changes should not affect any existing code using RichText.

## Issue ticket number and link


## Checklist before requesting a review
### For each PR
- [x] Add Copyright if it missed:   
      - `"Copyright (c) 2019-present Axmol Engine contributors (see AUTHORS.md)."`
- [x] I have performed a self-review of my code.
       
   Optional:
   - [ ] I have checked readme and add important infos to this PR.
   - [ ] I have added/adapted some tests too.
          
### For core/new feature PR
- [x] I have checked readme and add important infos to this PR.
- [x] I have added thorough tests.

rh101 committed:
Add paragraph tag support
Add HTML header tags support
Add multi-new line rich text element
Add support for end of tag action
General code clean up

rh101 committed:
Remove redundant additions

rh101 committed:
Fix compilation issue on platforms other than Windows

halx99 commented on core/ui/UIRichText.h:
```diff
@@ -64,14 +63,14 @@ class AX_GUI_DLL RichElement : public Ref
      * @lua NA
      */
-    virtual ~RichElement(){};
+    ~RichElement() override = default;
```
default destructor can be removed?

rh101 replied:
Yes, a few them like that in the code. I'll clean them all up if you don't mind.

rh101 committed:
Remove redundant constructors and destructors.
Initialize member variables.
Remove explicit inline specifier since all member class functions are implicitly inlined.
